plugins/services/src/js/components/modals/NewCreateServiceModalForm.js

```diff
@@ -2,81 +2,207 @@ import Ace from 'react-ace';
 import classNames from 'classnames';
 import React from 'react';
 
-import Tabs from '../../../../../../src/js/components/Tabs';
+import Batch from '../../../../../../src/js/structs/Batch';
+import JSONConfigReducers from '../../reducers/JSONConfigReducers';
+import JSONParserReducers from '../../reducers/JSONParserReducers';
+import {combineParsers} from '../../../../../../src/js/utils/ParserUtil';
+import {combineReducers} from '../../../../../../src/js/utils/ReducerUtil';
+import ServiceFormSection from '../forms/ServiceFormSection';
 import TabButton from '../../../../../../src/js/components/TabButton';
 import TabButtonList from '../../../../../../src/js/components/TabButtonList';
+import Tabs from '../../../../../../src/js/components/Tabs';
 import TabView from '../../../../../../src/js/components/TabView';
 import TabViewList from '../../../../../../src/js/components/TabViewList';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+import Util from '../../../../../../src/js/utils/Util';
+
+const METHODS_TO_BIND = [
+  'handleFormChange',
+  'handleFormBlur',
+  'handleJSONBlur',
+  'handleJSONChange'
+];
 
-const METHODS_TO_BIND = ['handleJSONChange'];
+const SECTIONS = [
+  ServiceFormSection
+];
+
+const jsonParserReducers = combineParsers(JSONParserReducers);
+const jsonConfigReducers = combineReducers(JSONConfigReducers);
+const inputConfigReducers = combineReducers(
+  Object.assign({}, ...SECTIONS.map((item) => item.configReducers))
+);
+const validationReducers = combineReducers(
+  Object.assign({}, ...SECTIONS.map((item) => item.validationReducers))
+);
 
 class NewCreateServiceModalForm extends React.Component {
   constructor() {
     super(...arguments);
 
+    let batch = new Batch();
+
     this.state = {
-      jsonValue: ''
+      appConfig: {},
+      batch,
+      errors: {},
+      jsonValue: JSON.stringify(batch.reduce(jsonConfigReducers, {}), null, 2)
     };
 
     METHODS_TO_BIND.forEach((method) => {
       this[method] = this[method].bind(this);
     });
   }
 
+  handleJSONBlur() {
+    let {errors, jsonValue} = this.state;
+    let newState = {};
+    let appConfig;
+
+    try {
+      appConfig = JSON.parse(jsonValue);
+    } catch (event) {
+      // TODO: handle error
+      newState.errors = Object.assign(errors, {
+        jsonEditor: 'JSON value is not valid json.'
+      });
+    }
+
+    if (appConfig) {
+      // Flush batch
+      let batch = new Batch();
+
+      // Run validation reducers on appConfig with new on minimal batch
```

Comment:
Could you rephrase this comment? I'm struggling to figure out what you mean by`with new on minimal batch`